pandas/core/series.py

```diff
@@ -299,7 +299,12 @@ def __init__(
             if index is None:
                 if not is_list_like(data):
                     data = [data]
-                index = ibase.default_index(len(data))
+
+                if is_empty_data(data):
+                    index = Index([])
+                else:
+                    index = ibase.default_index(len(data))
```

Comment:
Is there a reason to not put this logic into the `default_index` function? I think would be more logical there

Reply:
I also thought about it but was shying away from it as it'll impact dataframes as well then. I'll give it a shot if you think it's a good idea